crates/viewer/re_renderer/src/video/player.rs

```diff
@@ -193,48 +193,32 @@ impl VideoPlayer {
         re_tracing::profile_function!();
 
         // Some terminology:
-        //   - presentation timestamp = composition timestamp
+        //   - presentation timestamp (PTS) == composition timestamp
         //     = the time at which the frame should be shown
-        //   - decode timestamp
+        //   - decode timestamp (DTS)
         //     = determines the decoding order of samples
         //
         // Note: `decode <= composition` for any given sample.
         //       For some codecs & videos, the two timestamps are the same.
         // We must enqueue samples in decode order, but show them in composition order.
         // In the presence of b-frames this order may be different!
 
-        // 1. Find the latest sample where `decode_timestamp <= presentation_timestamp`.
-        //    Because `decode <= composition`, we never have to look further ahead in the
-        //    video than this.
-        let Some(decode_sample_idx) = latest_at_idx(
-            &self.data.samples,
-            |sample| sample.decode_timestamp,
-            &presentation_timestamp,
-        ) else {
-            return Err(VideoPlayerError::EmptyVideo);
-        };
-
-        // 2. Search _backwards_, starting at `decode_sample_idx`, looking for
-        //    the first sample where `sample.presentation_timestamp <= presentation_timestamp`.
-        //    This is the sample which when decoded will be presented at the timestamp the user requested.
-        let Some(requested_sample_idx) = self.data.samples[..=decode_sample_idx]
-            .iter()
-            .rposition(|sample| sample.presentation_timestamp <= presentation_timestamp)
+        // Find sample which when decoded will be presented at the timestamp the user requested.
+        let Some(requested_sample_idx) = self
+            .data
+            .latest_sample_index_at_presentation_timestamp(presentation_timestamp)
```

Comment:
Nit: this would be shorter as an `.ok_or(VideoPlayerError::EmptyVideo)?` instead of `if-else